documentation/developer_setup.md

```diff
@@ -0,0 +1,126 @@
+
+This client takes three parameters, the first is the service URL for the specific account. The second is an [`interface`](https://gobyexample.com/interfaces) which is a specific struct that has definitions for a certain set of methods. In the case of `azcore.Credential` the `AuthenticationPolicy(options AuthenticationPolicyOptions) Policy` method must be defined to be a valid interface. The final argument to methods that create clients or interact with the service should be a pointer to an `Options` parameter. Making this final parameter a pointer allows the customer to pass in `nil` if there are no specific options they want to change. The `Options` type should have a name that is intuitive to what the customer is trying to do, in this case `TableClientOptions`.
+
+### Defining Methods
+Defining a method follows the format:
+```golang
+func (m *<MyStruct>) MethodName(param1 param1Type, param2 param2Type) (ReturnType, ReturnType2) {
+
+}
+```
+The `(m *<MyStruct>)` portion is the "receiver". Methods can be defined for either pointer (with a `*`) or receiver (without a `*`) types. Pointer receivers will avoid copying types on method calls and allow the method to mutate the receiving struct. You should use pointer receivers wherever possible to limit memory copies.
+
+
+Both public and private methods can be declared on clients. Below is an example in the `aztables` package for a `Create` method on the `TableServiceClient`:
+```golang
+// Create creates a table with the specified name.
+func (t *TableServiceClient) Create(ctx context.Context, name string) (TableCreateResponse, error) {
+	resp, err := t.client.Create(ctx, TableProperties{&name}, new(TableCreateOptions), new(QueryOptions))
+	if err == nil {
+		tableResp := resp.(TableCreateResponse)
+		return tableResp, nil
+	}
+	return TableCreateResponse{}, err
+}
+```
+
+All methods that perform I/O of any kind, sleep, or perform a significant amount of CPU-bound work must have the first parameter be of type [`context.Context`][golang_context] which allows the customer to carry a deadline, cancellation signal, and other values across API boundaries. The remaining parameters should be parameters specific to that method. The return types for methods should be first a "Response" object and second an `error` object.
+
+## Write Tests
+
+Testing is built into the Go toolchain as well with the `testing` library. The testing infrastructure located in the `sdk/internal` directory takes care of generating recordings, establishing the mode a test is being run in (options are "recording", "playback", "live-no-playback"), and reading environment variables.
+
+A simple test for `aztables` is shown below:
+```golang
+
+import (
+	"os"
+
+	"github.com/testify/assert"
```

Comment:
This is what the Go package I use:

`github.com/stretchr/testify/assert`

(It might be that it works both ways)